### PR TITLE
chore: ESG tweaks

### PR DESCRIPTION
**Pull request by nsprenkle**
**TL;DR -** minor fixes and tweaks to ESG functionality

JIRA: [AU-444](https://openedx.atlassian.net/browse/AU-444) & [AU-367](https://openedx.atlassian.net/browse/AU-367)

**What changed?**

- Renames `submission_id` param to `submission_uuid` param for submission locking for consistency.
- Filters cancelled submissions from submission return list in ESG.

**Testing Instructions**

- Manually exercise with Postman & updated contracts.
- For an ORA with cancelled submissions, hit `list_staff_workflows` XBlock handler and verify it returns successfully (used to break due to cancelled submissions mapping issues).

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @edx/masters-devs-gta

### CODE REVIEW

**Review by leangseu-edx — APPROVED** [review]
LGTM 👍

**Review by jansenk — APPROVED** [review]
